daemon/endpoint.go

```diff
@@ -160,24 +159,26 @@ func (d *Daemon) createEndpoint(epTemplate *models.EndpointChangeRequest, id str
 		return PutEndpointIDFailedCode, err
 	}
 
-	// If the endpoint has no labels, give the endpoint a special identity with
-	// label reserved:init so we can generate a custom policy for it until we
-	// get its actual identity.
-	if len(addLabels) == 0 {
+	addLabels := labels.NewLabelsFromModel(lbls)
+
+	if len(addLabels) > 0 {
+		addLabels, _, ok := checkLabels(addLabels, nil)
+		if !ok {
+			return PutEndpointIDInvalidCode, fmt.Errorf("No valid label")
```

Comment:
`checkLabels()` performs this check:

```
if len(addLabels) == 0 && len(delLabels) == 0 {
    return nil, nil, false
```

This is impossible, since immediately above you are ensuring that the length of `addLabels` is > 0.

Reply:
`checkLabels` calls `FilterLabels`, which can return an empty label list and trigger that condition.